CHANGES.md

```diff
@@ -1,6 +1,470 @@
+  (see https://chapel-lang.org/docs/latest/usingchapel/chplenv.html#chpl-llvm)
+* added a new default value of `unset` for the `CHPL_LLVM` environment variable
+  (see https://chapel-lang.org/docs/latest/usingchapel/chplenv.html#chpl-llvm)
+* `CHPL_TARGET_COMPILER` can now be used to choose the LLVM or C-based back-end
+  (see https://chapel-lang.org/docs/main/technotes/llvm.html
+   and https://chapel-lang.org/docs/main/usingchapel/chplenv.html#chpl-compiler)
+* new variables `CC`/`CXX` support specifying the paths to C/C++ compilers
+  (see https://chapel-lang.org/docs/main/usingchapel/chplenv.html#chpl-compiler)
+* a C++14 compiler is now required to build the Chapel compiler
+  (see https://chapel-lang.org/docs/main/usingchapel/prereqs.html)
+* Python 3.7 is now required by tools like `c2chapel`, `chapeldoc`, etc.
+  (see https://chapel-lang.org/docs/main/usingchapel/prereqs.html)
+* added support for a new platform setting for HPE Apollo systems
+  (see TODO)
+* default to `fifo` tasking on arm-based (M1) Macs
+* replaced `CHPL_REGEXP=re2|none` with `CHPL_RE2=bundled|none`
+  (see https://chapel-lang.org/docs/main/usingchapel/chplenv.html#chpl-re2)
+* removed previously deprecated `CHPL_` env. settings that `bundled` replaced
+
+Semantic Changes / Changes to Chapel Language
+---------------------------------------------
+* started changing `.size` on ranges/domains/arrays to return `int` by default
+  (see https://chapel-lang.org/docs/1.25/builtins/ChapelRange.html#ChapelRange.range.size)
+* started changing `.indices` queries on arrays to differentiate from `.domain`
+  (see https://chapel-lang.org/docs/1.25/builtins/ChapelArray.html#ChapelArray.indices)
+
+Namespace Changes
+-----------------
+* made operations on `c_fn_ptr` types automatically available for now
+
+New Features
+------------
+* added a prototypical new `manage` statement for context management
+  (see TODO)
+* added a `foreach` loop for single-task parallel loops
+  (see https://chapel-lang.org/docs/1.25/technotes/foreach.html)
+* added support for `extern union` to refer to external unions in C
+  (see https://chapel-lang.org/docs/main/language/spec/interoperability.html#referring-to-external-c-structs-and-unions)
+* added `.sizeAs()` to query range/domain/array sizes using a specific type
+  (see https://chapel-lang.org/docs/1.25/builtins/ChapelRange.html#ChapelRange.range.sizeAs,
+   and https://chapel-lang.org/docs/1.25/builtins/ChapelArray.html#ChapelArray.sizeAs)
+
+Feature Improvements
+--------------------
+* added support for slicing an associative array by an associative domain
+* added support for ranges, domains, and arrays over 1-element enumerations
+  (e.g., `enum color { red }; var r: range(idxType=color);` now works)
+* added support for operators on scalar/tuple pairs in which coercion occurs
+  (e.g., `2 * (1.0, 2.0, 3.0)` now works, resulting in `(2.0, 4.0, 6.0)`)
+* various improvements to interfaces:
```

Comment:
I might suggest

```
* improved the prototype support for interfaces
  - improved the interfaces implementation to better match language design
    - disabled automatic inference of implements statements
    - added support for interface methods
    - added support for calling functions defined within an interface
    - added support for required functions to be interface-constrained
    - added support for invoking required functions via associated constraints
    - added support for passing arguments of associated types
  - fixed several problems when using interfaces with other language features
    - with `in` intents
    - with generic `implements` statements
    - when a function needs implicit conversions to satisfy a requirement
    - when a function needs promotion to satisfy a requirement
```

Here I think we could remove all of the 3rd-level bullets if you wanted to summarize more aggressively. I don't think the CHANGES file needs to have a ton of detail here because this is still a prototype, undocumented, unstable language feature.

Reply:
Thanks Michael!